setup/bindir/cloud-set-guest-password.in

```diff
@@ -26,25 +26,56 @@
 # Modify this line to specify the user (default is root)
 user=root
 
-# Add your DHCP lease folders here
-DHCP_FOLDERS="/var/lib/dhclient/* /var/lib/dhcp3/* /var/lib/dhcp/*"
-PASSWORD_SERVER_PORT=8080
-password_received=0
-file_count=0
-error_count=0
-
-for DHCP_FILE in $DHCP_FOLDERS; do
-	if [ -f $DHCP_FILE ]; then
-		file_count=$((file_count+1))
-		PASSWORD_SERVER_IP=$(grep dhcp-server-identifier $DHCP_FILE | tail -1 | awk '{print $NF}' | tr -d '\;')
-
-		if [ -n "$PASSWORD_SERVER_IP" ]; then
-			logger -t "cloud" "Found password server IP $PASSWORD_SERVER_IP in $DHCP_FILE"
-                        break
-		fi
-	fi
-done
+# Add network provider variables here (default means that interface for network manager is eth0)
+NETPLAN=/etc/netplan
+IFUPDOWN=/etc/network/interfaces
+NETMAN=/etc/sysconfig/network-scripts/ifcfg-eth0
+
+# Add dhcp variables
+PASSWORD_SERVER_PORT=8080                                                                                                                                                                                                                    password_received=0                                                                                                                                                                                                                          error_count=0
+
+# OS is using netplan
+if [ -d "$NETPLAN" ]; then
+        logger -t "cloud" "Operating System is using netplan"
+
+        PASSWORD_SERVER_IP=$(netplan ip leases eth0 | grep SERVER_ADDRESS | awk '{split($0,a,"="); print a[2]}')
+
+        if [ -n "$PASSWORD_SERVER_IP" ]; then
+                 logger -t "cloud" "Found password server IP $PASSWORD_SERVER_IP in netplan config"
+        fi
+fi
+
+# OS is using ifupdown
+if [ -f "$IFUPDOWN" ]; then
+        logger -t "cloud" "Operating System is using ifupdown"
+
+        DHCP_FOLDERS="/var/lib/dhclient/* /var/lib/dhcp3/* /var/lib/dhcp/*"
+        file_count=0
```

Comment:
Maybe moving this up outside of the if block?

Reply:
I have moved it outside of the if block. I had left it there initially because only in this loop will there be a loop and and a counter but no pb.